rfd/0219-k8s-rbac-simplification.md

```diff
@@ -0,0 +1,280 @@
+
+#### Helm Chart
+
+The Helm Chart will by default provision the new Kubernetes resources starting
+in v19.0.0. An option will be added to the chart to allow opting out of creating
+the default ClusterRoleBindings. All existing release branches will have this
+flag set to disable the new functionality to minimize breaking changes.
+
+
+Example:
+
+```yaml
+roles: kube,app,discovery
+authToken: foo
+proxyAddr: example.devteleport.com:443
+kubeClusterName: myCluster
+rbac:
+  defaultClusterRoleBindings: true
+```
+
+The ClusterRoleBindings created by the chart will be prefixed with the
+chart release and the namespace to allow multiple deployments in the
+same Kubernetes cluster not to conflict since ClusterRoleBinding resources
+are global and not namespaced.
+
+```yaml
+
+{{- if .Values.rbac.defaultClusterRoleBindings -}}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:metadata:
+  name: {{ .Release.Name }}-{{ .Release.Namespace }}-preset-edit
+```
+
+
+#### Provision Script
+
+For simplicity, the provision script will be updated to always create the
+Kubernetes resources, and will not provide a means for users to opt out of
+the new behavior.
+
+#### Auto Discovery
+
+##### EKS / AKS
+
+For both EKS and AKS, auto-discovery will be updated to create the expected
+RBAC resources.
+
+##### GKE
+
+For GKE, we will need to change the documented GCP IAM role to enable admin
+privileges.
+Teleport will attempt to impersonate a privileged group, but there is a risk
+GKE users will not be able to use the new preset roles out of the box.
+
+A special attention to error management will be required to help users
+understand the issue and how to resolve it, which can be easily done by
+running the provision script for example.
+
+#### Auto-update
+
+Auto-update will not attempt to apply the new model, with the assumption that
+existing setups are already working.
+
+### UX
+
+#### Exec confusion
+
+As the underlying Kubernetes RBAC gets discouraged, it will help with the
+confusion around _exec_ being allowed with a `get` verb is gone.
+Teleport uses a dedicated verb to control access to `exec`.
+
+#### Resource enrollment UI
+
+On the Web UI, the initial page generates a _Helm_ command line.
+
+After enrollment, a test page is shown, prompting for a `kubernetes_groups`
+value, which defaults to the user's trait and updates it if changed in this
+page.
+
+The following page is about testing the connection, and also have a form to
+populate a subset of the users/groups.
+
+The initial idea was to allow the user to update it's traits on the fly to not
+have to logout/log back in, however, as there is no mention of traits, and
+because the form is present on multiple pages with different behavior, and
+because it requires the user to initially already have a role assigned that
+contains the trait interpolation, we will remove both the 'Setup access' page
+and the 'users/groups' form from the 'Test connection' page.
```

Comment:
We might need to leave them for existing Teleport users who don't have the kube_user and kube_group traits. We can populate them by default though. @marcoandredinis is the owner of the discovery UI process and might have opinions on this.